OOPS4/src/assignment2/MatrixOperations.java

```diff
@@ -0,0 +1,77 @@
+package assignment2;
+
+import java.util.InputMismatchException;
+
+/**
+ * @Class MatrixOperations implements all matrix operations indicated in
+ *        interface methods: add, multiply, transpose
+ */
+public class MatrixOperations extends Matrix implements IMatrixOperations {
+	public MatrixOperations() {
+	}
+
+	public MatrixOperations(int row, int column) {
+		super(row, column);
+	}
+
+	/**
+	 * @Method addMatrix(Matrix to_add) add two matrices matrices must have same row
+	 *         and column size for addition
+	 * @return type = Matrix object, new added matrix
+	 */
+	public Matrix addMatrix(Matrix to_add) {
+		if (this == null | to_add == null)
+			throw new NullPointerException();
+
+		if (this.row_size != to_add.row_size | this.column_size != to_add.column_size) {
+			throw new InputMismatchException();
+		}
+
+		Matrix add_result = new Matrix(row_size, column_size);
+		for (int row = 0; row < row_size; row++) {
+			for (int column = 0; column < column_size; column++) {
+				add_result.matrix[row][column] = this.matrix[row][column] + to_add.matrix[row][column];
+			}
+		}
+
+		return add_result;
+	}
+
+	/**
+	 * @Method multiplyMatrix(Matrix to_multiply) multiply two matrices matrices
+	 *         must satisfy m1.column_no = m2.row_no condition to multiply
+	 * @return type = Matrix object, new multiplied matrix
+	 */
+	public Matrix multiplyMatrix(Matrix to_multiply) {
+		if (this.column_size != to_multiply.row_size)
```

Comment:
Put null check here like this:
if (this == null | to_multiply== null)
	throw new NullPointerException();